daemons/pacemakerd/pcmkd_corosync.c

```diff
@@ -59,13 +60,32 @@ pcmk_cfg_dispatch(gpointer user_data)
     return 0;
 }
 
+static gboolean
+cluster_reconnect_cb(gpointer data)
+{
+    if (cluster_connect_cfg()) {
+        mainloop_timer_stop(reconnect_timer);
+        mainloop_timer_del(reconnect_timer);
+        crm_warn("Cluster reconnect succeeded");
```

Comment:
mainloop_timer_del() will call mainloop_timer_stop() first, so we don't need to do that explicitly. Also let's set it to NULL too, just to be safe. Maybe cluster_disconnect_cfg() should call mainloop_timer_del() too if not null, so valgrind doesn't complain if we happen to exit after starting the timer (the mainloop will be gone by that point so it doesn't really matter ...)